tests/system/Database/Live/ForgeTest.php

```diff
@@ -1237,4 +1204,168 @@ public function testDropKey()
 
         $this->forge->dropTable('key_test_users', true);
     }
+
+    public function testAddTextColumnWithConstraint()
+    {
+        // some DBMS do not allow a constraint for type TEXT
+        $result = $this->forge->addColumn('user', [
+            'text_with_constraint' => ['type' => 'text', 'constraint' => 255, 'default' => ''],
+        ]);
+
+        $this->assertTrue($result);
+
+        // SQLSRV requires dropping default constraint before dropping column
+        $result = $this->forge->dropColumn('user', 'text_with_constraint');
+
+        $this->assertTrue($result);
+    }
+
```

Comment:
This is a really messy set of changes to a table. Add/drop columns and indexes. Lets test all our work and see if we can pass!